packages/mdc-fab/mdc-fab.scss

```diff
@@ -12,98 +12,42 @@
 
-  display: inline-flex;
-  position: relative;
-  justify-content: center;
-  width: 56px;
-  height: 56px;
-  padding: 0;
-  transition: box-shadow 280ms $mdc-animation-standard-curve-timing-function;
-  border: none;
-  border-radius: 50%;
-  cursor: pointer;
-  user-select: none;
-  box-sizing: border-box;
-  fill: currentColor;
-  -moz-appearance: none;
-  -webkit-appearance: none;
-  overflow: hidden;
-
-  &:not(.mdc-ripple-upgraded) {
-    -webkit-tap-highlight-color: rgba(0, 0, 0, .18);
-  }
-
-  @include mdc-theme-prop(background-color, secondary);
-  @include mdc-theme-prop(color, text-primary-on-secondary);
-  @include mdc-elevation(6);
-
-  &--mini {
-    width: 40px;
-    height: 40px;
-  }
-
-  &--plain {
-    background-color: white;
-
-    @include mdc-theme-prop(color, text-primary-on-light);
-    @include mdc-ripple-bg((pseudo: "::before"));
-    @include mdc-ripple-fg((pseudo: "::after"));
-  }
-
-  &:active,
-  &:focus {
-    outline: none;
-  }
-
-  &:active {
-    @include mdc-elevation(12);
-  }
-
-  &:hover {
-    cursor: pointer;
-  }
+#{$mdc-fab} {
+  @include mdc-fab-base;
+  @include mdc-fab-theme(map-get($mdc-fab-theme-config-defaults, standard));
+}
 
-  &::-moz-focus-inner {
-    padding: 0;
-    border: 0;
-  }
+#{$mdc-fab--plain} {
+  @include mdc-fab-theme(map-get($mdc-fab-theme-config-defaults, plain));
+}
 
-  // This allows for using SVGs within them to align properly in all browsers.
-  // Can remove once: https://bugzilla.mozilla.org/show_bug.cgi?id=1294515 is resolved.
+// This rule must appear in the source after all other modifier classes in order for its properties to take precedence
+// over them, because they set some of the same properties.
+#{$mdc-fab--mini} {
+  @include mdc-fab-theme--mini;
+}
 
-  // stylelint-disable selector-max-type
-  // postcss-bem-linter: ignore
-  > svg {
-    width: 100%;
-  }
-  // stylelint-enable selector-max-type
+// This rule must appear in the source after all other modifier classes in order for its properties to take precedence
+// over them, because they set some of the same properties.
+@include mdc-fab-selector-disabled(#{$mdc-fab--mini}) {
+  @include mdc-fab-theme--mini;
+}
 
-  fieldset:disabled &,
-  &:disabled {
-    background-color: rgba(0, 0, 0, .12);
-    color: rgba(0, 0, 0, .26);
-    cursor: default;
-    pointer-events: none;
-  }
+// This rule must appear in the source after the preceding classes in order for its properties to take precedence
+// over them, because they set some of the same properties.
+// TODO(acdvorak): Visually hide the FAB completely when disabled, rather than just changing its color.
+@include mdc-fab-selector-disabled(#{$mdc-fab}) {
+  @include mdc-fab-theme--disabled(map-get($mdc-fab-theme-config-defaults, disabled));
 }
 
-.mdc-fab__icon {
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  width: 100%;
+#{$mdc-fab__icon} {
```

Comment:
Hmmmm, it seems an old style problem we have. I have two questions here
1. Does this needed to be a mixin? I feel this is a bit overkill and I wonder if it might be better if it is part of the base style?
2. Could we nest this icon into fab instead of having it as standalone? To follow: `modifier` leaves outside but `element` lives nested.